### PR TITLE
Lightloop scalarization rebased on master

### PR DESCRIPTION
**Pull request by FrancescoC-unity**
Same as https://github.com/Unity-Technologies/ScriptableRenderPipeline/pull/2209 but manually rebased to master. 

Katana: https://katana.bf.unity3d.com/projects/com.unity.render-pipelines/builders?automation-tools_branch=add-platform-filter&ScriptableRenderLoop_branch=lightloop-scalarization&unity_branch=trunk fails on unrelated LWRP test, I am rekicking it to see if it is an instability issue.

### CODE REVIEW

File: com.unity.render-pipelines.high-definition/Runtime/Lighting/LightLoop/LightLoop.hlsl [hunk 1]
```diff
@@ -1,5 +1,7 @@
 #include "Packages/com.unity.render-pipelines.core/ShaderLibrary/Macros.hlsl"
 
+#define SCALARIZE_LIGHT_LOOP (defined(SUPPORTS_WAVE_INTRINSICS) && defined(LIGHTLOOP_TILE_PASS) && SHADERPASS == SHADERPASS_FORWARD)
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/Runtime/Lighting/LightLoop/LightLoop.hlsl]: Add a comment here to say that we only perform such optimsation for forward (i.e be a bit more versobse :) )

Comment (sebastienlagarde) [reply com.unity.render-pipelines.high-definition/Runtime/Lighting/LightLoop/LightLoop.hlsl]: + add a link to your blog post

File: com.unity.render-pipelines.high-definition/Runtime/Material/Decal/DecalUtilities.hlsl [hunk 2]
```diff
@@ -26,90 +35,151 @@ void ApplyBlendNormal(inout float4 dst, inout int matMask, float2 texCoords, int
...
+    float4x4 worldToDecal = ApplyCameraTranslationToInverseMatrix(decalData.worldToDecal);
+    float3 positionDS = mul(worldToDecal, float4(posInput.positionWS, 1.0)).xyz;
+    positionDS = positionDS * float3(1.0, -1.0, 1.0) + float3(0.5, 0.5f, 0.5);  // decal clip space
+    if ((all(positionDS.xyz > 0.0f) && all(1.0f - positionDS.xyz > 0.0f)))
+    {
+        float2 uvScale = float2(decalData.normalToWorld[3][0], decalData.normalToWorld[3][1]);
+        float2 uvBias = float2(decalData.normalToWorld[3][2], decalData.normalToWorld[3][3]);
+        positionDS.xz = positionDS.xz * uvScale + uvBias;
+        positionDS.xz = frac(positionDS.xz);
+        // clamp by half a texel to avoid sampling neighboring textures in the atlas
```
[the remark this answers is not in the document]
Comment (FrancescoC-unity) [reply com.unity.render-pipelines.high-definition/Runtime/Material/Decal/DecalUtilities.hlsl]: Will do, that was a straight copy paste from what it was before, but I def. agree this is a tad hard to read.

**Comment by sebastienlagarde** [comment]
>fails on unrelated LWRP test, I am rekicking it to see if it is an instability issue.
yes, this seems to fail on master. unrelated to this PR

**Pull request merged**